parser_test.go

```diff
@@ -0,0 +1,708 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lexparse
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ianlewis/runeio"
+)
+
+var (
+	errTreeMismatchSize  = errors.New("trees have different number of nodes")
+	errTreeMismatchValue = errors.New("trees node values do not match")
+)
+
+const (
+	debugPrint = false
+	inputABC   = "A B C"
+)
+
+// compareTrees returns true if two trees are equivalent by comparing the
+// value of each node in both trees.
+func compareTrees[T any](tr1, tr2 *Tree[T]) (bool, error) {
```

Comment:
Rather than walking the two trees separately it should probably just walk the two trees at the same time.
See this example for a binary tree: https://stackoverflow.com/questions/19310674/pseudocode-to-compare-two-trees

Reply:
I can change that.  Maybe tomorrow.